README.md

```diff
@@ -1,19 +1,186 @@
+Duvet annotations create a network effect.
+These annotations help give context to code
+especially if the code is not your preferred language.
+They link back to the design document
+to give background and more information.
+Since these annotations are bidirectional,
+are waypoints that let you move between
+the more abstract design document
+and the detailed implementation.
+
+![](./img/debugger.png)
+
+## When do you stop adding tests?
+
+But now we are done right?
+Every requirement has an implementations and a tests.
+They all have been reviewed.
+Duvet can only see that you TRIED to test.
+How does the reviewer know you tested ENOUGH?
+
+Duvet is a mechanism to connect the requirement to the source,
+but it leaves the quality and completeness of these links to humans.
+A tests makes sure that something is true once,
+it does not make sure that something is *always* true.
+Todays edge case it tomorrow's issue.
+
+## Dafny
+
+Dafny is a *verification-aware* programming language.
+
+Formal verification let's us translate the Duvet annotation
+into a specification that says
+"for every string in the *universe* this must **always** be true".
+The Dafny verifier will use deductive reasoning to say "Yup, that is true".
+This is like writing a test, but you are testing every value all at once
+statically before you run any code.
+
+Dafny uses formal logic to prove specifications.
+It doesn't forget about edge cases
+because it checks every case.
+
+Since is very useful to have verified code,
+but only if you can run it
+Dafny compiles to many other languages:
+Java, C#, JavaScript, Go, and more on the way!
+
+![](./img/dafny.png)
+
+## Mechanisms!
+
+Putting this together we have a strong chain
+from the designer's intent right to what code is executing.
+
+Now we have two mechanisms.
+First Duvet makes us write a specification for every requirement.
+Then Dafny proves that our implementation matches our specification.
+
+![](./img/mechanisms.png)
+
+## But what are we going to do in this workshop?
+
+We are going to build an Amazon Resource Name (ARN) parser.
+This may not seem that exciting,
+but it is a real problem that came up in writing the AWS Encryption SDK.
+It has a nice balance of complexity and simplicity.
+It has a lot of the same flavor of other more complex problems
+like serialization of binary formats.
+
+## Let’s start!
 
 Just follow along with [the workshop instructions](./instructions/steps.md) right here on GitHub!
 
+## To learn more
+
+* [Dafny](http://dafny.org/), a programming language that formally verifies your implementation matches your specification
+* [Duvet](https://github.com/awslabs/duvet), a code quality tool for measuring how well your human-readable specification is covered in code
+* AWS Encryption SDK in Dafny: [https://github\.com/aws/aws\-encryption\-sdk\-dafny](https://github.com/aws/aws-encryption-sdk-dafny)
+* s2n: [https://github\.com/aws/s2n\-tls](https://github.com/aws/s2n-tls)
```

Comment:
Yup, definitely mention these towards the top somewhere.

Reply:
moved